### PR TITLE
Components for EventLog and ExternalIntegrations

### PR DESCRIPTION
**Pull request by SzymonPobiega**
Adds `ServiceControlComponent`s for the event log and external integrations

### CODE REVIEW

**Comment by SzymonPobiega** [comment]
@timbussmann @cquirosj Pls review :)

**Review by mikeminutillo — COMMENTED** [review]
Overall, I like it. I'm not sure if EventLog should be a component as it is a cross-cutting concern. I don't know that we have a good definition for that though and this does help to create a stronger division between components.

**Review by timbussmann — APPROVED** [review]

File: src/ServiceControl/EventLog/EventLogMappings.cs [hunk 1]
```diff
@@ -2,13 +2,27 @@
...
-        public EventLogMappings(Dictionary<Type, Type> mappings)
+        public EventLogMappings(IEnumerable<IEventLogMappingDefinition> mappers)
         {
-            this.mappings = mappings;
+            foreach (var mapper in mappers)
+            {
+                var type = mapper.GetType();
+                if (type.BaseType == null)
+                {
+                    return;
```
[the remark this answers is not in the document]
Comment (SzymonPobiega) [reply src/ServiceControl/EventLog/EventLogMappings.cs:17]: Yeah. Definitely. I copied that code from `EventLogMappingsFeature` without looking at it :( Thanks!

Comment (SzymonPobiega) [inline src/ServiceControl/EventLog/EventLogMappings.cs:17]:
```suggestion
                    continue;
```

**Comment by timbussmann** [comment]
> I'm not sure if EventLog should be a component as it is a cross-cutting concern. I don't know that we have a good definition for that though and this does help to create a stronger division between components.

that's fair. We think removing the usage of NSB features adds value and will allow for easier refactoring if the decision will be to not treat them as dedicated components.

**Pull request merged**